capstone/src/main/java/com/google/sps/servlets/StudySetServlet.java

```diff
@@ -175,6 +189,131 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     String requestResultJSON = gson.toJson(getRequestResult(request));
     response.getWriter().println(requestResultJSON);
   }
-  //TODO doPost for creating and storing a study set
 
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+    throws IOException {
+    HashMap<String, Object> result_par = new Gson()
+    .fromJson(request.getReader(), HashMap.class);
+    String ownderID = result_par.get("user_id").toString();
+    String title = result_par.get("title").toString();
+    String subject = result_par.get("subject").toString();
+    String description = result_par.get("description").toString();
+    String university = result_par.get("university_id").toString();
+    String professor = result_par.get("professor").toString();
+    String academicTime = result_par.get("academic_time").toString();
+    String courseName = result_par.get("course_name").toString();
+
+    SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+    Date date = new Date();
+    String creationTime = formatter.format(date);
+    Timestamp creationTimestamp = Timestamp.valueOf(creationTime);
```

Comment:
Converting from Date to String to Timestamp is overly complicated. Try `new Timestamp(date.getTime())` and see if you get the same answer.